test/boost/reader_concurrency_semaphore_test.cc

```diff
@@ -863,7 +864,7 @@ SEASTAR_THREAD_TEST_CASE(test_reader_concurrency_semaphore_admission) {
         BOOST_REQUIRE_EQUAL(stats_after.reads_enqueued_for_admission, stats_before.reads_enqueued_for_admission + 1);
         BOOST_REQUIRE_EQUAL(semaphore.get_stats().waiters, 1);
 
-        std::ignore = post_enqueue_hook(cookie1);
+        [[maybe_unused]] auto value = post_enqueue_hook(cookie1);
```

Comment:
Why this change?

Reply:
Due to this test: https://github.com/scylladb/scylladb/blob/885e81bedc27b42138fcf838dbe234b46759198a/test/boost/reader_concurrency_semaphore_test.cc#L924-L932

and the change
https://github.com/scylladb/scylladb/pull/21918/files/885e81bedc27b42138fcf838dbe234b46759198a#diff-34cabee484b74a85b640a0fe07406cbfbce8a9722b4d3bd6bb1c639930495f36R1591

The `awaits_guard` returned from the second hook is created and destroyed if not assigned. Therefore, by the time `maybe_admit_waiters()` is called in the `execution_loop`, the permit would no longer be in the `active_await` state and as a consequence the second permit would be admitted.

The change, ensures the guard is not immediately destroyed keeping the first permit in the `active_await` state.

Reply:
I see. When looking not deeply enough, the two statements look similar, but std::ignore indeed causes immediate destruction.

Suggest to rename `value` to `guard` to hint to readers about what's going on.